### ISSUE TITLE
Unable to load model trained on multi-gpu machine, on single-gpu machine, 

### ISSUE THREAD

**Issue by kyoungrok0517** [dmlc/xgboost#4361]
I see the following error if I use below code to load the model. The model was trained on Linux multi-gpu machine. I'm trying to load the model on Windows single-gpu machine.

```
XGBoostError: b'[12:21:26] c:\\users\\administrator\\desktop\\xgboost\\src\\common\\common.h:182: Check failed: n_gpus <= n_available_devices (1 vs. 0) Starting from gpu id: 1, there are only 0 available devices, while n_gpus is set to: 1'
```

**[Code]**
```python
with open('./model/model_skipgram_2019-04-04T05-30-20.pkl', 'rb') as f:
    model = pickle.load(f)
```
**[Model]**
Model ([link](https://www.dropbox.com/s/iba259mom5m8usd/model_skipgram_2019-04-04T05-30-20.pkl?dl=0))

**Comment by kyoungrok0517** [comment]
If I do `model.set_param('gpu_id', 0)` on multi-gpu machine then load on the single-gpu machine I could load the model. I think there's some bug in sanity checking code.

**Comment by trivialfis** [comment]
@hcho3 Can we talk about our progress on Json at some point or give a roadmap on this?  I don't want to touch the parameter configuration before having a flexible model format  ..

**Comment by hcho3** [comment]
@trivialfis I recall that parameter validation should be tied together with the JSON serialization. Did you get a chance to review the complete schema? Feel free to post at the JSON RFC thread if you have more detailed idea.

**Comment by hcho3** [comment]
@trivialfis Right now, I am reviewing your CMake refactor PR and also revamping CI.

**Comment by kyoungrok0517** [comment]
Any update on this issue??

**Comment by hcho3** [comment]
@kyoungrok0517 I just tried loading the pickle you uploaded on p2.xlarge machine (single GPU), and I did not get the same error.

**Comment by trivialfis** [comment]
closed by #4454 .
